pandas/tools/merge.py

```diff
@@ -199,10 +204,53 @@ def get_result(self):
         typ = self.left._constructor
         result = typ(result_data).__finalize__(self, method='merge')
 
+        if self.indicator:
+            result = self._indicator_post_merge(result, indicator_string)
+
         self._maybe_add_join_keys(result, left_indexer, right_indexer)
 
         return result
 
+    def _indicator_pre_merge(self, left, right, indicator_arg):
+
```

Comment:
don't pass ``indicator_arg``. instead in the initializetion stage, just define (e.g. when the ``_Merge`` is defined
``self.indicator_name = self.indicator if isinstance(self.indicator, compat.string_type) else '_merge')``